MatomoTracker/EventAPISerializer.swift

```diff
@@ -91,10 +91,10 @@ fileprivate extension Event {
                 URLQueryItem(name: "c_i", value: contentInteraction),
                 
                 URLQueryItem(name: "idgoal", value: goalId != nil ? "\(goalId!)" : nil),
-                URLQueryItem(name: "revenue", value: revenue != nil ? "\(revenue!)" : nil),
+                // The revenue parameter can be set via the revenue of a goal or the revenue of an order
+                URLQueryItem(name: "revenue", value: (revenue ?? orderRevenue) != nil ? "\((revenue ?? orderRevenue)!)" : nil),
 
                 URLQueryItem(name: "ec_id", value: orderId),
-                URLQueryItem(name: "revenue", value: orderRevenue != nil ? "\(orderRevenue!)" : nil),
```

Comment:
This line here was the culprit. We were always setting the `revenue` parameter again based on the value of an order. In the case of a goal, the order revenue is always `nil` and thus it removes the `revenue` of the order.